Add section on extending code and instances (#868)
stellar/stellar-docs: docs/learn/encyclopedia/storage/state-archival.mdx

env.storage().temporary().get(&EXAMPLE_KEY); // Returns Ok(2)
```

> All `Instance` storage is stored in a single contract instance `LedgerEntry` and shares a single TTL. This means that one call to `env.storage().instance().extend_ttl()` will extend the TTL of all `Instance` entries, as well as the contract instance itself. `env.storage().instance().extend_ttl()` will also extend the contract code entry as well, which we'll expand on in the next section. For `Temporary` and `Persistent` storage, each entry has its own TTL and must be extended individually. The interface is slightly different and takes the key of the entry being extended as well as the TTL extension value.

Reviewer: ```suggestion
All `Instance` storage is stored in a single contract instance `LedgerEntry` and shares a single TTL. This means that one call to `env.storage().instance().extend_ttl()` will extend the TTL of all `Instance` entries, as well as the contract instance itself. It will also extend the contract code entry as well, which we'll expand on in the next section. For `Temporary` and `Persistent` storage, each entry has its own TTL and must be extended individually. The interface is slightly different and takes the key of the entry being extended as well as the TTL extension value.
```

A call to `extend_ttl(N)` ensures that the current TTL of the contract instance entry is _at least_ N ledgers. For example, if `extend_ttl(100)` is called and the contract instance entry has a current TTL of 50 ledgers, the TTL will be extended up to 100 ledgers. If `extend_ttl(100)` is called and the contract instance entry has a current TTL of 150 ledgers, the TTL will not be extended and the `extend_ttl()` call is a no-op.

In addition to contract-defined TTL extensions using the `extend_ttl()` function, a contract data entry's TTL can be extended via the [`ExtendFootprintTTLOp`](#extendfootprintttlop) operation.

## Contract Code and Contract Instance lifetimes

> Contract code entries and contract instances also have lifetimes, and there are a couple ways to extend them. Methods like `env.storage().instance().extend_ttl()` and `env.deployer().extend_ttl()` extend **both** the contract code and contract instance. Note that the threshold check and extensions are done for both the contract code and contract instance, so it’s possible that one is bumped but not the other depending on what the current TTL’s are.

Reviewer: ```suggestion
Contract code entries and contract instances have lifetimes, and there are a couple ways to extend them. Methods like `env.storage().instance().extend_ttl()` and `env.deployer().extend_ttl()` extend both the contract code and contract instance. The threshold check and extensions are done independently for both the contract code and contract instance, so it’s possible that one is bumped but not the other depending on what the current TTL’s are.
```

> If you would like to extend the contract code or contract instance separately, you can use `env.deployer().extend_ttl_for_code()` and `env.deployer().extend_ttl_for_contract_instance()` respectively. Calling both `env.deployer().extend_ttl_for_code()` and `env.deployer().extend_ttl_for_contract_instance()` with the same parameters is equivalent to calling only `env.deployer().extend_ttl()`.

Reviewer: ```suggestion
If you would like to extend the contract code or contract instance separately, you can use `env.deployer().extend_ttl_for_code()` and `env.deployer().extend_ttl_for_contract_instance()` respectively. Calling both with the same parameters is equivalent to calling only `env.deployer().extend_ttl()`.
```
